Define O_FSYNC in libconstants.gen.c

Musl libc prior to c1f4ed150137d793c9d07356305a89e8785e7e02 has a bug
in which it uses O_FSYNC without defining it anywhere.  Work around it
by explicitly defining O_FSYNC as O_SYNC as musl does with that
commit.
---
 gen_constants.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gen_constants.sh b/gen_constants.sh
index 0f330b9..6191d7d 100755
--- a/gen_constants.sh
+++ b/gen_constants.sh
@@ -58,6 +58,10 @@ cat <<-EOF > "${OUTFILE}"
 /* GENERATED BY MAKEFILE */
 $INCLUDES
 
+#ifndef O_FSYNC
+#define O_FSYNC O_SYNC
+#endif
+
 #include "libconstants.h"
 const struct constant_entry constant_table[] = {
 $(echo "$INCLUDES" | \
-- 
2.10.1

